src/hardwaresecuritymodules/azext_hardwaresecuritymodules/generated/_params.py

```diff
@@ -0,0 +1,65 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-statements
+
+from knack.arguments import CLIArgumentType
+from azure.cli.core.commands.parameters import (
+    tags_type,
+    resource_group_name_type,
+    get_location_type
+)
+from azure.cli.core.commands.validators import get_default_location_from_resource_group
+from azext_hardwaresecuritymodules.action import (
+    AddNetworkProfileSubnet,
+    AddNetworkProfileNetworkInterfaces
+)
+
+
+def load_arguments(self, _):
+
+    with self.argument_context('hardwaresecuritymodules dedicated-hsm list') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('top', help='Maximum number of results to return.')
+
+    with self.argument_context('hardwaresecuritymodules dedicated-hsm show') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('name', options_list=[
+                   '--name', '-n'], help='The name of the dedicated HSM.')
+
+    with self.argument_context('hardwaresecuritymodules dedicated-hsm create') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('name', options_list=[
+                   '--name', '-n'], help='Name of the dedicated Hsm')
+        c.argument('location', arg_type=get_location_type(self.cli_ctx),
+                   validator=get_default_location_from_resource_group)
+        c.argument('sku', type=str,
+                   help='The HSM device SKU if a non-standard HSM is wanted (default is: SafeNet Luna Network HSM A790)')
+        c.argument('zones', nargs='+',
+                   help='The Dedicated Hsm zones.')
+        c.argument('tags', tags_type)
+        c.argument(
+            'stamp_id', help='This field will be used when RP does not support Availability zones.')
+        c.argument('network_profile_subnet', options_list=['--network-profile-subnet', '-s'], action=AddNetworkProfileSubnet, nargs='+', help='Specifies the identifier '
+                   'of the subnet. Expected value: id=xx.')
```

Comment:
You can use `arg_group='Network Profile'` and make the option to be `--subnet/-s`. Similar changes for `network_profile_network_interfaces`.

Reply:
Following up with service team to confirm network interfaces should not be an array.

Reply:
After talking to the service team, they would have to change the service side to change this from an array. They do not want to do this now since they will be adding a new SKU (HSM device) that will require some swagger rework in anyway in the next couple months. They will change it during that work.

Reply:
I can switch to arg groups, though